src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -346,16 +347,16 @@ export class SharedCredentialsProvider implements CredentialsProvider {
             }
         }
 
+        const profiles = toRecord(this.allSharedCredentialProfiles.keys(), k => this.allSharedCredentialProfiles.get(k))
+
         return fromIni({
             profile: this.profileName,
             mfaCodeProvider: async mfaSerial => await getMfaTokenFromUser(mfaSerial, this.profileName),
             roleAssumer: assumeRole,
-            loadedConfig: loadedCreds
-                ? Promise.resolve({
-                      credentialsFile: loadedCreds,
-                      configFile: {},
-                  } as SharedConfigFiles)
-                : undefined,
+            loadedConfig: Promise.resolve({
+                credentialsFile: loadedCreds ?? profiles,
```

Comment:
Fixes a discrepancy (bug) with the SDK. They don't merge profiles but we do.

Reply:
worth adding as a code comment?